boards/posix/native_sim/doc/index.rst

```diff
@@ -70,38 +167,419 @@ You can use this target if you cannot compile or run 32 bit binaries.
+
+**Interrupt controller**:
+  A simple yet generic interrupt controller is provided. It can nest interrupts
+  and provides interrupt priorities. Interrupts can be individually masked or
+  unmasked. SW interrupts are also supported.
+
+**Clock, timer and system tick model**
+  This model provides the system tick timer. By default
+  :kconfig:option:`CONFIG_SYS_CLOCK_TICKS_PER_SEC` configures it to tick every 10ms.
+
+  Please refer to the section `About time in native_sim`_ for more
+  information.
+
+**UART/Serial**
+   Two optional native UART drivers are available:
+
+   **PTTY driver (UART_NATIVE_POSIX)**
+      With this driver, one or two Zephyr UART devices can be created. These
+      can be connected to the Linux process stdin/stdout or a newly created
+      pseudo-tty. For more information refer to the section `PTTY UART`_.
+
+   **TTY driver (UART_NATIVE_TTY)**
+      An UART driver for interacting with host-attached serial port devices
+      (eg. USB to UART dongles). For more information refer to the section
+      `TTY UART`_.
+
+**Real time clock**
+  The real time clock model provides a model of a constantly powered clock.
+  By default this is initialized to the host time at boot.
+
+  This RTC can also be set to start from time 0 with the ``--rtc-reset`` command
+  line option.
+
+  It is possible to offset the RTC clock value at boot with the
+  ``--rtc-offset=<offset>`` option,
+  or to adjust it dynamically with the function :c:func:`native_rtc_offset`.
+
+  After start, this RTC advances with the simulated time, and is therefore
+  affected by the simulated time speed ratio.
+  See `About time in native_sim`_ for more information.
+
+  The time can be queried with the functions :c:func:`native_rtc_gettime_us`
+  and :c:func:`native_rtc_gettime`. Both accept as parameter the clock source:
+
+  - ``RTC_CLOCK_BOOT``: It counts the simulated time passed since boot.
+    It is not subject to offset adjustments
+  - ``RTC_CLOCK_REALTIME``: RTC persistent time. It is affected by
+    offset adjustments.
+  - ``RTC_CLOCK_PSEUDOHOSTREALTIME``: A version of the real host time,
+    as if the host was also affected by the clock speed ratio and offset
+    adjustments performed to the simulated clock and this RTC. Normally
+    this value will be a couple of hundredths of microseconds ahead of the
+    simulated time, depending on the host execution speed.
+    This clock source should be used with care, as depending on the actual
+    execution speed of native_sim and the host load,
+    it may return a value considerably ahead of the simulated time.
+
+**Entropy device**:
+  An entropy device based on the host :c:func:`random` API.
+  This device will generate the same sequence of random numbers if initialized
+  with the same random seed.
+  You can change this random seed value by using the command line option:
+  ``--seed=<random_seed>`` where the value specified is a 32-bit integer
```

Comment:
Nit, and additional Sphinx trivia, you can do:

```suggestion
  :samp:`--seed={<random_seed>}` where the value specified is a 32-bit integer
```

There's a few other places in the PR where this could be leverages, but 100% understand if you want to leave this for a later update :)